### PR TITLE
feat: cosmos tx history support

### PR DESCRIPTION
**Pull request by pastaghost**

### CODE REVIEW

File: packages/chain-adapters/package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     "type-check": "tsc --project ./tsconfig.json --noEmit"
   },
   "dependencies": {
+    "@shapeshiftoss/unchained-tx-parser": "^5.4.0",
```
Comment (kaladinlight) [inline packages/chain-adapters/package.json:28]:
```suggestion
```

File: packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts [hunk 2]
```diff
@@ -90,9 +91,44 @@ export abstract class CosmosSdkBaseAdapter<T extends CosmosChainTypes> implement
...
+
+          return {
+            address: input.pubkey,
+            blockHash: parsedTx.blockHash,
+            blockHeight: parsedTx.blockHeight,
+            blockTime: parsedTx.blockTime,
+            caip2: this.getCaip2(),
+            chain: this.getType(),
+            confirmations: parsedTx.confirmations,
+            txid: tx.txid,
```
Comment (kaladinlight) [inline packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts:112]: might as well use `parsedTx` data throughout.
```suggestion
            txid: parsedTx.txid,
```

File: packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts [hunk 3]
```diff
@@ -90,9 +91,44 @@ export abstract class CosmosSdkBaseAdapter<T extends CosmosChainTypes> implement
...
+            fee: parsedTx.fee,
+            status: getStatus(parsedTx.status),
+            tradeDetails: parsedTx.trade,
+            transfers: parsedTx.transfers.map<chainAdapters.TxTransfer>((transfer) => ({
```
Comment (kaladinlight) [inline packages/chain-adapters/src/cosmossdk/CosmosSdkBaseAdapter.ts:116]: I think I went a little overboard with the explicit type casting if we want to remove some characters
```suggestion
            transfers: parsedTx.transfers.map((transfer) => ({
```

File: packages/types/src/chain-adapters/index.ts [hunk 4]
```diff
@@ -31,29 +31,14 @@ export type AssetBalance = {
   caip19: string
 }
 
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
 type ChainSpecificTransaction<T> = ChainSpecific<
   T,
   {
     [ChainTypes.Bitcoin]: bitcoin.TransactionSpecific
   }
 >
```
Comment (kaladinlight) [inline packages/types/src/chain-adapters/index.ts:41]: no sense in keeping this around until if/when we even need it
```suggestion
```

File: packages/types/src/chain-adapters/index.ts [hunk 5]
```diff
@@ -174,9 +159,7 @@ export type SubscribeError = {
 }
 
 export type TxHistoryResponse<T extends ChainTypes> = {
-  page: number
-  totalPages: number
-  txs: number
+  cursor: string
```
Comment (kaladinlight) [inline packages/types/src/chain-adapters/index.ts:162]: Might be worth returning `pubkey` as well like we do on `getAccount`

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**